### PR TITLE
Enhancement - Case insensitive provider name 

### LINKED ISSUE #159: Provider name should not be case sensitive
### Scenario/Intent
Specify my provider in a case insensitive way.

### Environment Details
 - **Puppet module for HPE OneView Version:** 2.2.1

### Steps to Reproduce
Specify "hardware_variant" as all lowercase or mixed case in my login.json file, for instance:
```json
  "hardware_variant": "c7000"
  "hardware_variant": "synergy"
  "hardware_variant": "sYnErGy"
```
### Expected Result
Case should not matter as long as the hardware variant specifies a valid type, such as 'c7000' or 'synergy'.

### Actual Result
It requires specifically 'C7000' or 'Synergy' to be specified.

### PR DESCRIPTION
**Pull request by fgbulsoni**
### Description
Allows a provider to be specified in a case insensitive manner.
i.e.: `"C7000"`, `"c7000"`, `"Synergy"`, `"synergy"`, `"sYnErGy"`

EDIT: Also allows for symbols to be passed in (case insensitive as well).
i.e.: `:c7000`, `:synergy`, `:SynErgy`, `:C7000`

### Issues Resolved
#159 

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass (`$ rake test`).
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: lib/puppet/provider/login.rb [hunk 1]
```diff
@@ -26,6 +26,7 @@ def login
   }
   credentials = load_authentication_settings(options)
   credentials[:hardware_variant] ||= 'C7000'
+  credentials[:hardware_variant] = credentials[:hardware_variant].capitalize
```
Comment (tmiotto) [inline lib/puppet/provider/login.rb]: Please use `capitalize!` instead.

**Comment by tmiotto** [comment]
By any means the hardware_variant could be a symbol?

**Comment by fgbulsoni** [comment]
> By any means the hardware_variant could be a symbol?

Shouldn't. At least at the moment. 

All the providers are either confined to `'C7000'` or `'Synergy'` at the moment, with a code like:
```ruby
confine true: login[:hardware_variant] == 'Synergy'
```

The way to accept symbols that would cause the least changes is probably to just turn the `login[:hardware_variant]` into a string and capitalize it inside the login.rb file.

Something like:
```ruby
credentials[:hardware_variant] = credentials[:hardware_variant].to_s.capitalize
```

Looks acceptable? We'd lose the `.capitalize!` but would be able to accept `symbols`

**Comment by fgbulsoni** [comment]
For now, I've gone with 

```ruby
credentials[:hardware_variant] = credentials[:hardware_variant].to_s.capitalize
```

So we can also accept symbols besides strings for synergy and c7000.

The tests had also failed with capitalize! due to ENV variables being considered as frozen, so assigning a variable here would be necessary either way to support both json and ENV variables.

@tmiotto please review again :octocat:

**Review by tmiotto — APPROVED** [review]
Seems pretty good.

🍰 ✨

**Pull request merged**